src/os/bluestore/BlueStore.cc

```diff
@@ -4740,14 +4740,17 @@ int BlueStore::mkfs()
         if (r < 0) {
           derr << __func__ << " fsck found fatal error: " << cpp_strerror(r)
                << dendl;
-          return r;
         }
         if (r > 0) {
           derr << __func__ << " fsck found " << r << " errors" << dendl;
           r = -EIO;
         }
       }
-      return r; // idempotent
+
+      if (r == 0)
+        return 0; // idempotent
+
```

Comment:
this is not correct, if fsck fails, we will have return values != 0 and we proceed with next steps?? Don't we need to fail here? I don't understand the logic clearly here.  we can zap/format the disk and recreate the osd or am i missing something here?